### ISSUE TITLE
Expose function for manually framing a signal

### ISSUE THREAD

**Issue by kdonbekci** [pytorch/audio#1395]
## 🚀 Feature
Expose function for manually framing a signal/waveform in a consistent manner with existing transforms.

## Motivation
Transforms such as `ta.transforms.Spectrogram` handle windowing with overlap under the hood. Sometimes, it is necessary to achieve the same windowing behavior for other input types that will not undergo a FFT. Currently, there seems to be no API exposed for manually windowing a signal, akin to TensorFlow's `tf.signal.frame` function. 

## Pitch
A function such as `torchaudio.frame(x, win_length, hop_length, padding_args)` that matches the behavior of the implemented transforms. 

## Alternatives
External libraries like `librosa` have functionality for framing an arbitrary signal with overlap, however to match the default behavior of PyTorch Audio, the user needs to pad the signal beforehand, causing extra code and a potential source of bugs. 

PyTorch Core has the `.unfold` method, but again the padding considerations need to be handled manually.

**Comment by kdonbekci** [comment]
Looks like `F.unfold` alone can match the current behavior of Torch Audio transforms besides the `"reflect"` padding option.

**Comment by mthrok** [comment]
Hi @kdonbekci 

Thanks for the idea. I think this is a helpful addition as I often see the need to align processed signal and the original signal. (cc @faroit I think this is your interest too)

I looked at the documentation of `tf.signal.frame` and my understanding is that the proposed feature is folding the original signal accommodating the overlaps caused by the hop length. (correct me if I am wrong.)

Can you illustrate the end-to-end example of how you would use the resulting frames, detailing what you expect in `matches the behavior of the implemented transforms.`

For example applying Spectrogram on single channel signal (shape: `[1, num_frames]`) gives spectrogram with shape `[1, num_fft_bins, num_steps]`. In this case the `win_length` would not show up explicitly as output shape. The number of FFT bins is free variable, if I understand correctly. The resulting `num_steps` would be related to `hop_length`, `win_length` and `padding`.

```python
import torch
import torchaudio.transforms as T

waveform = torch.randn(1, 1024)
transform = T.Spectrogram(n_fft=400)
spec = transform(waveform)

print(waveform.shape)
print(spec.shape)

>>> torch.Size([1, 1024])  # channel, num frames
>>> torch.Size([1, 201, 6])  # channel, fft bins, steps
```
